graphql/src/main/resources/graphql-schema/baseSchema.graphql

```diff
@@ -0,0 +1,5 @@
+schema {
+    query: Query
+}
+
+type Query {}
```

Comment:
A question as GraphQL newbie. Does it really make sense to have a kind of base schema here and all services extending this? For example, if we need a `mutation`, should we add that here as empty and then extending in each service (i.e. KafkaAdminService) schema definition? I am wondering how much sense has this base schema here.

Reply:
Name can be missleading. The way I understand it this is entire schema and the way to extend it later will be to add more items to it etc.. mentioned mutation etc. Maybe there is some pattern I'm not aware of?

Reply:
There are modular schema libraries (or code first frameworks that generate schemas) out there but basically they defeat the purpose of the schemas themselves (as one place to define all operations for the server)

Reply:
@ppatierno as @wtrocki said, it is an empty schema and it will be extended later by adding more items. The graphql schema language allows you to break the schema up into smaller chunks and the individual chunks can extend types that are defined elsewhere. The reason that there is no mutation in that base schema is because the code would fail as you cannot have a mutation without fields. That is enforced by graphql-java and it looks like it has been tightened up between v14.0 and v16.0 so the original definition I created had an empty mutation but I had to remove it to stop a runtime error. If we added a mutation then you could add it back.

Reply:
Yes. As follow up PR we can add some basic schema and generate stubs to implement later (so we will have schema - resolvers 1:1 connection) using something like https://graphql-code-generator.com/docs/plugins/java

Reason for getting schema out there faster with stubs is that UI team could mock things out of the box without waiting for the implementation.